components/collection/utils/useCollectionDetails.ts

```diff
@@ -0,0 +1,100 @@
+import { NFTListSold } from './../../identity/utils/useIdentity'
+import { NFT } from '@/components/rmrk/service/scheme'
+import { getVolume } from '@/utils/math'
+
+type Stats = {
+  listedCount?: number
+  collectionLength?: number
+  collectionFloorPrice?: number
+  uniqueOwnersPercent?: string
+  collectionDailyTradedVolumeNumber?: bigint
+}
+
+const differentOwner = (nft: {
+  issuer: string
+  currentOwner: string
+}): boolean => {
+  return nft.currentOwner !== nft.issuer
+}
+
+export const useCollectionDetails = ({ collectionId }) => {
+  const { data } = useGraphql({
+    queryPrefix: 'subsquid',
+    queryName: 'collectionStatsById',
+    variables: {
+      id: collectionId,
+    },
+  })
+  const stats = ref<Stats>({})
+
+  watch(data, () => {
+    if (data.value) {
+      const uniqueOwnerCount = [
+        ...new Set(data.value.stats.base.map((item) => item.currentOwner)),
+      ].length
+
+      const differentOwnerCount =
+        data.value.stats.base.filter(differentOwner).length
+      stats.value = {
+        listedCount: data.value.stats.listed.length,
+        collectionLength: data.value.stats.base.length,
+        collectionFloorPrice: Math.min(
+          ...data.value.stats.listed.map((item) => parseInt(item.price))
+        ),
+        uniqueOwnersPercent: `${
+          (uniqueOwnerCount / differentOwnerCount) * 100
+        }%`,
+        collectionDailyTradedVolumeNumber: getVolume(
+          data.value.stats.sales.map((nft) => nft.events).flat()
```

Comment:
The result is not equal to the `DailyTradedVolume`? It's the total traded volume as I see in the graphql query.

`components/rmrk/Gallery/CollectionActivity.vue` the method at  line183  would help.

Reply:
updated